code/segmentedtickerplant/datastripe.q

```diff
@@ -3,12 +3,17 @@
 //calls segmenting.csv which customer edits as to how many segments there will be
 //calls filtermap.csv which customer edits as to what filters should be applied to each segment
 
-//function transfoms segmenting.csv into table format so it can be accessed
-//function transforms filtermap.csv into table format then into a mapping of wcRef to filter which can be accessed and applied to data
+//configload transfoms segmenting.csv into table format so it can be accessed
+//configload transforms filtermap.csv into table format then into a mapping of wcRef to filter which can be accessed and applied to data
+
+//creating empty versions of segmenting table and filter mapping so we can revert to default mode if issues with either csv
+
+.stpps.segmentconfig:([] table:();segmentID:();wcRef:());
+.stpps.segmentfiltermap:(!/)([] wcRef:();filter:())`wcRef`filter;
```

Comment:
```suggestion
.stpps.segmentfiltermap:exec wcRef!filter from ([] wcRef:();filter:());
```

IMO this is clearer - though for an empty version could also just define an empty dict, would give the same result:

```suggestion
.stpps.segmentfiltermap:()!();
```